src/hotspot/share/opto/c2compiler.hpp

```diff
@@ -61,13 +61,11 @@ class C2Compiler : public AbstractCompiler {
   // possible for only a limited set of available intrinsics whereas
   // a non-virtual dispatch is possible for all available intrinsics.)
   // Return false otherwise.
-  virtual bool is_intrinsic_supported(const methodHandle& method) {
-    return is_intrinsic_supported(method, false);
-  }
+  virtual bool is_intrinsic_supported(const methodHandle& method);
 
   // Check if the compiler supports an intrinsic for 'method' given the
   // the dispatch mode specified by the 'is_virtual' parameter.
-  bool is_intrinsic_supported(const methodHandle& method, bool is_virtual);
+  bool is_virtual_intrinsic_supported(vmIntrinsics::ID id, bool is_virtual);
```

Comment:
I find the new name of the method confusing because it suggests that the intrinsic is always virtual. Can we keep the old name?

Reply:
We still have `is_intrinsic_supported()` declared at line 64. New `is_virtual_intrinsic_supported()` is added to handle only virtual intrinsics`_hashCode` and `_clone`.  I simply moved these intrinsics handling into separate method because it is used only by C2 in `library_call.cpp`.

Reply:
Okay, got it. Thanks for the explanation.

Reply:
Okay but then why do we still need the parameter `is_virtual`?

Reply:
@dholmes-ora  I can remove parameter if I modify caller code like next:
```
-    is_available = compiler != NULL && compiler->is_intrinsic_available(mh, C->directive()) &&
-                   compiler->is_virtual_intrinsic_supported(id, is_virtual);
+    is_available = compiler != NULL && compiler->is_intrinsic_available(mh, C->directive());
+    if (is_available && is_virtual) {
+      is_available = compiler->is_virtual_intrinsic_supported(id);
+    }
```
Will it satisfy you?

Reply:
How many callers are there? From an API design perspective this method is either only for virtual intrinsics, so no parameter needed, or it is for general intrinsics and the parameter indicates what type.

Reply:
This is the only place where it is called.

Reply:
Actually I found that I don't need `is_virtual_intrinsic_supported()` because the same information provides existing [vmIntrinsics::does_virtual_dispatch(id)](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/classfile/vmIntrinsics.cpp#L173)